lib/whitehall_importer/integrity_checker/change_history_check.rb

```diff
@@ -0,0 +1,40 @@
+module WhitehallImporter
+  class IntegrityChecker::ChangeHistoryCheck
+    attr_reader :edition, :proposed_change_history, :publishing_api_change_history
+
+    def initialize(edition, proposed_change_history, publishing_api_change_history)
+      @edition = edition
+      @proposed_change_history = proposed_change_history
+      @publishing_api_change_history = publishing_api_change_history
+    end
+
+    def match?
+      return false unless compare_length_of_histories
+
+      if edition.draft?
+        compare_histories_excluding_first_timestamp
+      else
+        compare_histories(proposed_change_history, publishing_api_change_history)
+      end
+    end
+
+  private
+
+    def compare_length_of_histories
+      proposed_change_history.length == publishing_api_change_history.length
+    end
+
+    def compare_histories(proposed, publishing_api)
+      proposed.zip(publishing_api).all? do |proposed_history, publishing_api_history|
+        proposed_history[:note] == publishing_api_history[:note] && proposed_history[:public_timestamp] == publishing_api_history[:public_timestamp]
+      end
+    end
+
+    def compare_histories_excluding_first_timestamp
+      proposed_change_history_first_item, *proposed_change_history_items = proposed_change_history
+      publishing_api_change_history_first_item, *publishing_api_change_history_items = publishing_api_change_history
+
+      proposed_change_history_first_item[:note] == publishing_api_change_history_first_item[:note] && compare_histories(proposed_change_history_items, publishing_api_change_history_items)
```

Comment:
To make this a little more readable we could do something like:
```suggestion
      first_note_matches = proposed_change_history_first_item[:note] == publishing_api_change_history_first_item[:note]
      remaining_history_matches = compare_histories(proposed_change_history_items, publishing_api_change_history_items)
      first_note_matches && remaining_history_matches
```